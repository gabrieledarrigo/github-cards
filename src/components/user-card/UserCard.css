.UserCard {
  border: 1px solid #ededed;
  box-shadow: rgba(0, 0, 0, 0.3) 0px 0px 10px;
}

.UserCard-header {
  display: flex;
  padding: 1rem 1rem 0;
}

.UserCard-header-content {
  margin: 0 0 0 0.5rem;
}

.UserCard-name {
  font-size: var(--font-size-4);
  margin: 0;
}

.UserCard-login {
  font-size: var(--font-size-5);
  margin: 0;
}

.UserCard-company {
  font-size: var(--font-size-6);
  margin: 0;
}

.UserCard-bio {
  margin: 0;
}

.UserCard-figure {
  margin: 0;
}

.UserCard-figure > img {
  max-width: 6rem;
}

.UserCard-content {
  padding: 1rem 1rem 0;
}

.UserCard-stats {
  border-top: 1px solid #ededed;
  display: flex;
  list-style: none;
  margin: 1rem auto 0;
  padding: 0;
}

.UserCard-stats-item {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 5rem;
  flex: 1;
  justify-content: center;
  text-align: center;
}

.UserCard-stats-item:nth-child(2) {
  border-left: 1px solid #ededed;
  border-right: 1px solid #ededed;
}
