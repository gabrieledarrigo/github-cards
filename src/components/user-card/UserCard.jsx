/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import styles from './UserCard.module.css';

export default function UserCard({
  name,
  login,
  company,
  avatarUrl,
  bio,
  followers,
  publicGists,
  publicRepos,
}) {
  const goToProfilePage = useCallback(() => {
    window.location.assign(`https://www.github.com/${login}`);
  }, [login]);

  return (
    <article
      className={styles.userCard}
      onClick={goToProfilePage}
    >
      <header className={styles.header}>
        <figure className={styles.figure}>
          <img
            src={avatarUrl}
            title={`${name} avatar`}
            alt={`${name} avatar`}
          />
        </figure>

        <div className={styles.headerContent}>
          <h2 className={styles.name}>
            {name}
          </h2>
          <h3 className={styles.login}>
            {login}
          </h3>
          {company && (
            <h4 className={styles.company}>
              {company}
            </h4>
          )}
        </div>
      </header>

      {bio && (
        <div className={styles.content}>
          <p className={styles.bio}>
            {bio}
          </p>
        </div>
      )}

      <ul className={styles.stats}>
        <li className={styles.statsItem}>
          <strong className={styles.statsValue}>
            {followers}
          </strong>
          <div className={styles.statsName}>
            Followers
          </div>
        </li>
        <li className={styles.statsItem}>
          <strong className={styles.statsValue}>
            {publicGists}
          </strong>
          <div className={styles.statsName}>
            Gists
          </div>
        </li>
        <li className={styles.statsItem}>
          <strong className={styles.statsValue}>
            {publicRepos}
          </strong>
          <div className={styles.statsName}>
            Repositories
          </div>
        </li>
      </ul>
    </article>
  );
}

UserCard.propTypes = {
  name: PropTypes.string.isRequired,
  login: PropTypes.string.isRequired,
  company: PropTypes.string,
  avatarUrl: PropTypes.string.isRequired,
  bio: PropTypes.string,
  followers: PropTypes.number.isRequired,
  publicGists: PropTypes.number.isRequired,
  publicRepos: PropTypes.number.isRequired,
};

UserCard.defaultProps = {
  company: null,
  bio: null,
};
