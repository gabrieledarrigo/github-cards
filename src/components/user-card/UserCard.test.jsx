import React from 'react';
import renderer from 'react-test-renderer';
import UserCard from './UserCard';

describe('UserCard', () => {
  const props = {
    name: 'Gabriele D\'Arrigo',
    login: 'gabrieledarrigo',
    company: 'MotorK / @ripliveit ',
    avatarUrl: 'https://avatars1.githubusercontent.com/u/1985555?v=4',
    bio: 'I\'m a guy (:robot:) with a strong passion for programming and cutting-edge web technologies! I :heart: comics, cinema, sports, video games and my little Debby!',
    followers: 26,
    publicGists: 12,
    publicRepos: 30,
  };

  it('should render correctly', () => {
    const tree = renderer.create(
      <UserCard {...props} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
