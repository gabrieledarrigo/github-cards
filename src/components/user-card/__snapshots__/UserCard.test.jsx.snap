// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should render correctly 1`] = `
<article
  className="userCard"
>
  <header
    className="header"
  >
    <figure
      className="figure"
    >
      <img
        alt="Gabriele D'Arrigo avatar"
        src="https://avatars1.githubusercontent.com/u/1985555?v=4"
        title="Gabriele D'Arrigo avatar"
      />
    </figure>
    <div
      className="headerContent"
    >
      <h2
        className="name"
      >
        Gabriele D'Arrigo
      </h2>
      <h3
        className="login"
      >
        gabrieledarrigo
      </h3>
      <h4
        className="company"
      >
        MotorK / @ripliveit 
      </h4>
    </div>
  </header>
  <div
    className="content"
  >
    <p
      className="bio"
    >
      I'm a guy (:robot:) with a strong passion for programming and cutting-edge web technologies!
    </p>
  </div>
  <ul
    className="stats"
  >
    <li
      className="statsItem"
    >
      <strong
        className="statsValue"
      >
        26
      </strong>
      <div
        className="statsName"
      >
        Followers
      </div>
    </li>
    <li
      className="statsItem"
    >
      <strong
        className="statsValue"
      >
        12
      </strong>
      <div
        className="statsName"
      >
        Gists
      </div>
    </li>
    <li
      className="statsItem"
    >
      <strong
        className="statsValue"
      >
        30
      </strong>
      <div
        className="statsName"
      >
        Repositories
      </div>
    </li>
  </ul>
</article>
`;
